api/client/samples/analogous_years/lib/final_ranks_computation.py

```diff
@@ -78,17 +78,16 @@ def time_series(client, data_series, initial_date, final_date):
     :return: A dataframe with data from the relevant dates
     """
     logger = client.get_logger()
-    entities = {'metric_id', 'item_id', 'region_id', 'source_id', 'frequency_id', 'start_date'}
+    entities = {'metric_id', 'item_id', 'region_id', 'source_id', 'frequency_id', 'start_date_bound'}
     discard = []
     for entity_type in data_series:
         if entity_type not in entities:
             discard.append(entity_type)
     for entity_type in discard:
         data_series.pop(entity_type)
```

Comment:
the above 6 lines feel a bit wordy and non-pythonic. what about just `data_series = {k: data_series[k] for k in data_series if k in entities}` if that is what you are trying to do here?

Reply:
Thanks! Done!